### PR TITLE
Read from std::in when filename is "-" (hyphen)

### PR DESCRIPTION
**Pull request by sashafrey**

### CODE REVIEW

File: src/artm/core/collection_parser.cc [hunk 1]
```diff
@@ -24,6 +24,26 @@
 
 using boost::iostreams::mapped_file_source;
 
+namespace {
+class ifstream_or_cin {
```
Comment (romovpa) [inline src/artm/core/collection_parser.cc]: Hey, what the difference between `ifstream` and `cin`?
It seemed to me that `cin` is a special instance of `ifstream` (or maybe just `istream`). In unix systems there are a special filenames for cin/cout/cerr: `/dev/stdin`, ...

**Comment by romovpa** [comment]
travis have failed

**Comment by sashafrey** [comment]
ci is hopefully fixed - I forget to include <iostream>

std::cin and std::cout are part of the C++ standard
http://www.cplusplus.com/reference/iostream/cin/
They are cross-platform, while /dev/stdin won't work on windows.

**Comment by sashafrey** [comment]
``ifstream`` is an input stream class to operate on files
http://www.cplusplus.com/reference/fstream/ifstream/
The problem is that standard input is not a file, so I'm not sure ``ifstream`` can open it (unless OS provide a special hook for it like ``/dev/stdin``

**Comment by romovpa** [comment]
:ok_hand:

**Pull request merged**